### PR TITLE
changed ENV time parameters to float

### PR DESCRIPTION
**Pull request by friedolino78**

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
@friedolino78 Is this PR ready to review, or do you want to add more commits?

**Comment by friedolino78** [comment]
> @friedolino78 Is this PR ready to review, or do you want to add more commits?

it was discussed in: [https://sourceforge.net/p/zynaddsubfx/patches/40/](https://sourceforge.net/p/zynaddsubfx/patches/40/)

unit test needs to be fitted. i will have a look at it.

**Comment by friedolino78** [comment]
ready for review, i guess :-)

File: src/Params/EnvelopeParams.cpp [hunk 1]
```diff
@@ -114,8 +158,27 @@ static const rtosc::Ports localPorts = {
         d.reply(d.loc, "i", env->Envmode);
         rEnd},
 
-    {"envdt", rDoc("Envelope Delay Times"), NULL,
+    {"envdt", rDoc("Envelope Delay Times (log)"), NULL,
```
Comment (fundamental) [inline src/Params/EnvelopeParams.cpp]: ms, not log, right?

File: src/Params/EnvelopeParams.cpp [hunk 2]
```diff
@@ -114,8 +158,27 @@ static const rtosc::Ports localPorts = {
...
+                arg_types[i] = 'f';
+            }
+            d.replyArray(d.loc, arg_types, args);
+        } else {
+            for(int i=0; i<N && i<M; ++i) {
+                env->envdt[i] = (rtosc_argument(msg, i).f)/1000; //store as seconds in member variable
+            }
+        }
+        rEnd},
+    {"dt", rDoc("Envelope Delay Times (lin)"), NULL,
```
Comment (fundamental) [inline src/Params/EnvelopeParams.cpp]: seconds not linear, right?

File: src/Params/EnvelopeParams.cpp [hunk 3]
```diff
@@ -496,8 +555,15 @@ void EnvelopeParams::getfromXML(XMLwrapper& xml)
     for(int i = 0; i < Penvpoints; ++i) {
         if(xml.enterbranch("POINT", i) == 0)
             continue;
-        if(i != 0)
-            Penvdt[i] = xml.getpar127("dt", Penvdt[i]);
+        if(i != 0) {
+            if(!xml.hasparreal("A_dt")) {
```
Comment (fundamental) [inline src/Params/EnvelopeParams.cpp]: This seems to be the only actual bug. It should be xml.hasparreal("dt") unless I'm mistaken.

**Pull request merged**